src/views/Header.js

```diff
@@ -209,11 +219,32 @@ class Header extends React.Component<void, HeaderProps, void> {
     }
 
     const pointerEvents = offset !== 0 || isStale ? 'none' : 'box-none';
+
+    // Only measure `title` component
+    const onLayout = name === 'title'
+      ? (e: LayoutEvent) => {
+        this.setState({
+          widths: {
+            ...this.state.widths,
+            [`${index}`]: e.nativeEvent.layout.width,
+          },
+        });
+      }
+      : undefined;
+
+    const width = name === 'left' || name === 'right'
+      ? this.state.widths[`${index}`]
+      : undefined;
+
     return (
       <Animated.View
         pointerEvents={pointerEvents}
+        onLayout={onLayout}
         key={`${name}_${key}`}
         style={[
+          width && {
```

Comment:
With mode `float`, we render 2 scenes ahead https://github.com/react-community/react-navigation/blob/c58a99712172cfe09c10ddb52a21249fdc52222f/src/views/Header.js#L202 so that the animation is smooth. That should make it possible to calculate the widths appropriately as far as I understand.